extension/src/service/web-component-service.js

```diff
@@ -113,10 +113,17 @@ function extendRequestObjWithApplicationInfo(requestObj) {
 }
 
 function extendRequestObjWithMetadata(requestObj, commercetoolsProjectKey) {
-  requestObj.metadata = {
-    // metadata key must have length of max. 20 chars
-    // metadata value must have length of max. 80 chars
-    ctProjectKey: commercetoolsProjectKey,
+  if (requestObj.metadata) {
+    requestObj.metadata = {
+      ...requestObj.metadata,
```

Comment:
It is allowed to pass only 20 metadata: https://docs.adyen.com/api-explorer/#/CheckoutService/v68/post/payments__reqParam_metadata. Should we already make a check here and reject the request immediately?

Reply:
I would not validate - we can not handle all the possible Adyen errors in the integration. We just need to make sure that our generated payload is valid but user specific data should be validated by Adyen unless it poses an issue for the integration itself, which is not the case here, right?